l1-contracts/contracts/upgrades/L1GatewayBase.sol

```diff
@@ -69,11 +81,31 @@ abstract contract L1GatewayBase {
         return abi.encode(additionalForceDeploymentsData);
     }
 
-    function getTokenName(address _token) external view returns (string memory) {
-        return IERC20Metadata(_token).name();
-    }
+    /// @notice Calls a token's metadata method.
+    /// @dev For the sake of simplicity, we expect that either of the
+    /// following is true:
+    /// 1. The token does not support metadata methods
+    /// 2. The token supports it and returns a `bytes32` string there.
+    /// 3. The token supports it and returns a correct `string` as a returndata.
+    ///
+    /// For all other cases, this function will panic and so such chains would not be
+    /// deployable.
+    function _safeCallTokenMetadata(address _token, bytes memory data) internal view returns (string memory, bool) {
+        // We are not afraid if token returns large calldata, since it affects
+        // only the deployment of the chain that uses such a malicious token.
```

Comment:
Are there any existing chains that have "malicious" tokens? Will we not brick them?

Reply:
no